examples/proxy/envelope-encryption/PREPARE_KMS.md

```diff
@@ -0,0 +1,127 @@
+# Preparing a Key Management Service (KMS)
+
+As discussed [here](./README.md) you must provide a KMS solution in order to use the Envelope Encryption Filter.
+For Tech Preview, only HashiCorp Vault is supported.
+
+## Preparing HashiCorp Vault
+
+You may either use an existing instance of HashiCorp Vault you have available in your organisation, or deploy a development instance
+HashoCorp Vault to your OpenShift Cluster which is good enough to try out the Filter.
+
+## Using an Existing Vault Instance
+
+### Prerequistes
+
+You must have an HashCorp Vault instance running on premise or you have a HashiCorp HCP Vault or Enterprise.  Free tier HCP Vault Secrets is
+*not* adequate.
+
+* You must have adminstrative access to the Vault instance or Vault Namespace.
+* You must know the Address of the Vault Server, and if Vault Enterprise, the name of the Vault Namespace(s).
+* Vault CLI installed.
+* GNU sed
+
+Note: these examples are prepared against HashiCorp Vault 1.15.
+
+### Configure the Existing Instance
+
+1. Set the `VAULT_ADDR` (and `VAULT_NAMESPACE` if Enterprise) to refer to the Vault Environment.
+   ```sh
+   export VAULT_ADDR=https://<vault server>:8200
+   export VAULT_NAMESPACE=<namespace(s)>
+2. Login to Vault as the Adminstrator.
+   ```sh
+   vault login
+   ```
+3. Enable the Transit Secrets Engine.
+   ```sh
+   vault secrets enable transit
+   ```
+   If the Transit engine was already enabled, an error will be printed but this can be ignored.
+4. Create a variable containing the Vault Transit Url.  This will be used later on.
+   ```sh
+      export VAULT_TRANSIT_URL=${VAULT_ADDR}/v1/${VAULT_NAMESPACE}/transit
+   ```
+5. Update the proxy config to refer to your Vault instance:
+   ```sh
+      sed -i "s/\(vaultTransitEngineUrl:\).*$/\1 ${VAULT_TRANSIT_URL}/" */proxy/proxy-config.yaml
+   ```  
+6. To conclude follow the steps below to [Create the Vault Policy and Filter Vault Token](#create-the-vault-policy-and-filter-vault-token)
+
+## Deploying a standalone development instance of Vault
+
+This option will give you an development instance of HashiCorp Vault with ephemeral storage.  This is sufficient to help you evaluate the
+Filter, but is *not* suitable for production or controlled test environments.
+
+It is assumed that you'll be deploying the HashiCorp Vault feature to the same OpenShift Cluster where you'll be evaluating AMQ Stream Proxy.
```

Comment:
```suggestion
It is assumed that you'll be deploying the HashiCorp Vault feature to the same OpenShift cluster where you'll be evaluating AMQ Stream Proxy.
```

Reply:
In Streams the _house style_ is we don't capitalize things like _cluster_ only proper nouns like Openshift. I don't know if we want to go along with that?  I won't continue suggesting in this review.